### PR TITLE
Fix CI build break (passed PR)

### PR DESCRIPTION
**Pull request by heaths**

### CODE REVIEW

**Comment by heaths** [comment]
@chidozieononiwu @weshaggard any idea why this passed PR? It was a warning (fixing it is non-trivial and affects shared source), but passed the PR build of the ci.yml pipeline def, only to fail in the rolling CI.

File: sdk/keyvault/samples/sharelink/ShareLink.csproj [hunk 1]
```diff
@@ -4,6 +4,10 @@
     <OutputType>Exe</OutputType>
     <TargetFramework>netcoreapp3.1</TargetFramework>
     <RootNamespace>Azure.Security.KeyVault.Storage</RootNamespace>
+    <NoWarn>
+      $(NoWarn);
```
Comment (pakrym) [inline sdk/keyvault/samples/sharelink/ShareLink.csproj:8]: Hm. We don't suppress this anywhere else.

**Review by pakrym — APPROVED** [review]

File: sdk/keyvault/samples/sharelink/ShareLink.csproj (hunk 1 above)
Comment (heaths) [reply sdk/keyvault/samples/sharelink/ShareLink.csproj:8]: True, and I looked into that as well. Basically, by samples being in the language repo, they pick up some globals (properties, targets, etc.) while not picking up others that match a more strict patter - a big reason I didn't want to include samples. In this case, though, I needed to link or duplicate source. Even in another repo, I imagine you'd agree that duplicating Azure.Core shared source is not ideal.

Comment (pakrym) [reply sdk/keyvault/samples/sharelink/ShareLink.csproj:8]: Yes, but there is nothing that disables this warning to pick up.

Comment (heaths) [reply sdk/keyvault/samples/sharelink/ShareLink.csproj:8]: https://github.com/Azure/azure-sdk-for-net/issues/14563

Comment (pakrym) [reply sdk/keyvault/samples/sharelink/ShareLink.csproj:8]: I bet it's the netcoreapp3.1 target.

**Pull request merged**